app/selectors/autocomplete.js

```diff
@@ -242,3 +242,22 @@ export const makeGetMatchTermForDateMention = () => {
         return lastMatchTerm;
     };
 };
+
+export const getDeletedPublicChannelsIds = createSelector(
+    getMyChannels,
+    getOtherChannels,
+    (myChannels, otherChannels) => {
+        const channels = myChannels.filter((c) => {
+            return (c.type === General.OPEN_CHANNEL);
+        }).concat(otherChannels);
+
+        return new Set(channels.filter((c) => c.delete_at !== 0).map((c) => c.id));
+    }
+);
+
+export const getMyChannelIdMemberships = createSelector(
```

Comment:
Why use a selector to convert this to a set? and why does it have to be a set?

Reply:
If it's just to make checking membership easier, you can also use the object version and just check `myMembers[userId]` instead of `myMembers.has(userId)`